src/client/options/mod.rs

```diff
@@ -560,6 +573,9 @@ impl ClientOptions {
                 .resolve_client_options(&options.hosts[0].hostname)
                 .await?;
 
+            // Save the original SRV hostname to allow mongos polling.
+            options.original_srv_hostname = Some(options.hosts[0].hostname.clone());
```

Comment:
Cloning here isn't strictly necessary since we're about to overwrite the old hosts, but we'd have to use something like `into_iter().next().unwrap()` to avoid it, and I think the marginal performance cost is worth the readability.